Xi/xiproperty.c

```diff
@@ -293,7 +293,7 @@ get_property(ClientPtr client, DeviceIntPtr dev, Atom property, Atom type,
 
     /* Return type, format, value to client */
     n = (prop_value->format / 8) * prop_value->size;    /* size (bytes) of prop */
-    ind = offset << 2;
+    ind = (unsigned long)offset << 2;
```

Comment:
The two times I see `get_property()` called in this file, values passed in for `offset` and `length` are effectively `uint32_t`, yet parameter types here are `int`. Might be worth changing that as well, so that you're only casting to a [potentially] larger unsigned type, not doing signed->unsigned conversion?

Reply:
> The two times I see `get_property()` called in this file, values passed in for `offset` and `length` are effectively `uint32_t`, yet parameter types here are `int`. Might be worth changing that as well, so that you're only casting to a [potentially] larger unsigned type, not doing signed->unsigned conversion?

Look, you're right, I'd rather do that. my typo

Reply:
done https://github.com/X11Libre/xserver/pull/22/commits/af939456c9224b4e815f4414a429b3f471309feb

Reply:
if `ind` is used there the first time, maybe also moving its declaration down there ?

Reply:
> if `ind` is used there the first time, maybe also moving its declaration down there ?

I didn't understand, you can suggest change directly on Github via webinterface and I can immediately include it in PR as commit.